runnable/extra-files/dexcept.h

```diff
@@ -0,0 +1,63 @@
+// D exceptions to by caught by C++
+
+#pragma once
+
+namespace D {
+
+struct string
+{
+    size_t length;
+    const char* ptr;
+};
+
+class Object;
+
+extern "C" {
+    string _d_toString(Object* o);
+}
+
+class Object
+{
+    void* __monitor;
+
+    virtual void _classinfo_data() = 0;
+
+protected:
+    // don't call directly, ABI mismatch

```

Comment:
> Do you think this alternative C++ mangling + _d_toString should be merged into druntime as-is? Or is it meant as prototype for a standardized upstream approach?

I'm not sure. This possibility is currently very specific to the WinEH implementation, I doubt it is feasible in dmd. I don't know if something similar would be possible for other platforms.